app/scripts/lib/app-start.js

```diff
@@ -757,6 +760,26 @@ define(function (require, exports, module) {
       return new Metrics(options);
     },
 
+    /**
+     * This function addresses the special scenario for Safari iOS
+     * where users using Private Browsing cannot verify their email
+     * (during sign up or sign-in confirmation) due to the
+     * 'window.localStorage' block.
+     *
+     * Returns `true` if the current browser is Safari iOS and the
+     * route is a verification email route.
+     *
+     * @returns {boolean}
+     * @private
+     */
+    _isVerificationInMobileSafari () {
+      const path = this._window.location.pathname;
+      const isVerificationPath = path === '/complete_signin' || path === '/verify_email';
```

Comment:
![](http://v14d.com/i/584044b7ad0da.png)

It is the 5% of errors. hmm, the problem with that one is that it gives you the password form and then logs you into settings. maybe it is worth living that one out for now.